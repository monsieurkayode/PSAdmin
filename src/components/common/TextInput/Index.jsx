import React from 'react';
import { string, func, bool } from 'prop-types';
import Loader from '../Loader/Index';

/**
 * @function TextInput
 *
 * @param {object} props
 *
 * @returns {JSX}
 */
const TextInput = ({
  name,
  label,
  placeholder,
  onChange,
  onFocus,
  value,
  error,
  loading
}) => {
  let wrapperClass = 'form-group';
  if (error && error.length > 0) {
    wrapperClass = `${wrapperClass} has-error`;
  }

  return (
    <div className={wrapperClass}>
      <label htmlFor={name}>{label}</label>
      <div className="field custom-field">
        <input
          type="text"
          name={name}
          className="form-control"
          placeholder={placeholder}
          onChange={onChange}
          onFocus={onFocus}
          value={value}
        />
        {loading && <Loader size={20} />}
      </div>
      {error && (
        <div
          style={{ maxWidth: '93%', padding: '5px' }}
          className="alert alert-danger"
        >
          {error}
        </div>
      )}
    </div>
  );
};

TextInput.defaultProps = {
  placeholder: '',
  value: '',
  error: ''
};

TextInput.propTypes = {
  name: string.isRequired,
  label: string.isRequired,
  onChange: func.isRequired,
  onFocus: func.isRequired,
  placeholder: string,
  value: string,
  error: string,
  loading: bool.isRequired
};

export default TextInput;
