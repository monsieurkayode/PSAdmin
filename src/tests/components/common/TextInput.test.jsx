import React from 'react';
import { mount } from 'enzyme';
import TextInput from '../../../components/common/TextInput/Index';

let wrapper;

const setup = (loading, error) => {
  const props = {
    name: '',
    label: '',
    value: '',
    placeholder: '',
    onChange: jest.fn(),
    onFocus: jest.fn(),
    error,
    loading
  };

  return mount(<TextInput {...props} />);
};

describe('TextInput Component', () => {
  it('renders correctly', () => {
    wrapper = setup(false, '');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a label and input element', () => {
    expect(wrapper.find('label').length).toEqual(1);
    expect(wrapper.find('input').length).toEqual(1);
  });

  it('renders a loader component when making async requests', () => {
    wrapper = setup(true, '');
    expect(wrapper.find('Loader').length).toEqual(1);
  });

  it('does not render loader component when not making async requests', () => {
    wrapper = setup(false, '');
    expect(wrapper.find('Loader').length).toEqual(0);
  });

  it('renders an error div when there are errors', () => {
    wrapper = setup(false, 'Invalid input length');
    expect(wrapper.find('.alert').length).toEqual(1);
  });
});
