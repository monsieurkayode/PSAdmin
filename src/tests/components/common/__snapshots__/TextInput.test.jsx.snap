// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Component renders correctly 1`] = `
<TextInput
  error=""
  label=""
  loading={false}
  name=""
  onChange={[MockFunction]}
  onFocus={[MockFunction]}
  placeholder=""
  value=""
>
  <div
    className="form-group"
  >
    <label
      htmlFor=""
    />
    <div
      className="field custom-field"
    >
      <input
        className="form-control"
        name=""
        onChange={[MockFunction]}
        onFocus={[MockFunction]}
        placeholder=""
        type="text"
        value=""
      />
    </div>
  </div>
</TextInput>
`;
